### DISCUSSION TITLE
How to specify testpaths that are sub-selected, not overridden, by paths specified on command line?

### DISCUSSION THREAD

**Discussion by MetRonnie** [pytest-dev/pytest#9136] [Q&A]
With a project structured like this:
```
project
|-- foo
|    `-- bar
|        `-- <Python source code>
|-- tests
|   |-- functional
|   |   `-- <Non-python tests>
|   |-- integration
|   |   `-- <Python tests>
|   `-- unit
|       `-- <Python tests>
`-- pytest.ini
```
We want to only ever run tests in these 3 directories specified by `testpaths`:
```ini
[pytest]
addopts =
    --doctest-modules
testpaths =
    foo/bar/  # Doctests
    tests/unit/
    tests/integration/
```
This works fine invoking `pytest` without specifying any paths on the command line. But tools like the test explorer in VSCode will specify paths when invoking `pytest` (e.g. `pytest /path/to/project` or `pytest .tests/`), and these paths cause `testpaths` to be ignored. Pytest then gets `ModuleNotFoundErrors` and so on because it's looking in `tests/functional`.

Essentially I am looking for a way to constrain the paths that Pytest uses to those contained within the paths specified in `pytest.ini`, but `testpaths` does not do this.

**Comment by RonnyPfannschmidt** [comment]
its a bug in the ide integration to pass in such paths by default

testpaths are the go-to for when no path is goven

 any passed in path will overide them with that path

**Reply by MetRonnie** [reply]
Though it seems to me there is a use case for being able to do `pytest tests/` and it only running `tests/unit` and `tests/integration` (if not using `testpaths`, then perhaps a new setting)?

**Comment by nicoddemus** [comment]
Alternatively you can exclude certain directories so pytest will never collect from them by using [collect_ignore_glob](https://docs.pytest.org/en/stable/reference.html#globalvar-collect_ignore_glob) in the root `conftest.py` file.
